### PR TITLE
Make version store optional for compiler

### PR DESCRIPTION
**Pull request by mcdurdin**

### CODE REVIEW

File: windows/src/developer/kmcmpdll/versioning.cpp [hunk 1]
```diff
@@ -1,45 +1,36 @@
...
-	FVersionInfo.MinVersion = version;
+  return TRUE;
 }
 
-void UpdateKeyboardVersion(PWSTR p)
-{
+BOOL VerifyKeyboardVersionF(PFILE_KEYBOARD fk, DWORD ver) {
+  if (fk->dwFlags & KF_AUTOMATICVERSION) {
+    fk->version = max(fk->version, ver);
+    return TRUE;
```
Comment (jahorton) [inline windows/src/developer/kmcmpdll/versioning.cpp]: Making sure I get this - this function is called at multiple points during compilation, allowing this function to incrementally compare its current minimum version against that needed to support each feature being checked if `KF_AUTOMATICVERSION` is specified.  If insufficient, the version is increased, resulting in the minimum necessary version being held in `fk->version` post-compilation.

File: windows/src/developer/kmcmpdll/Compiler.cpp [hunk 2]
```diff
@@ -559,7 +560,7 @@ BOOL CompileKeyboardHandle(HANDLE hInfile, PFILE_KEYBOARD fk)
 
 	delete str;
 
-	return TRUE;
+  return CheckKeyboardFinalVersion(fk);
```
Comment (jahorton) [inline windows/src/developer/kmcmpdll/Compiler.cpp]: This line sets the file's version at the very end of compilation?

File: windows/src/developer/kmcmpdll/versioning.cpp (hunk 1 above)
Comment (mcdurdin) [reply windows/src/developer/kmcmpdll/versioning.cpp]: Correct

File: windows/src/developer/kmcmpdll/Compiler.cpp (hunk 2 above)
Comment (mcdurdin) [reply windows/src/developer/kmcmpdll/Compiler.cpp]: This just prints an info report on the minimum version requirement that was determined by the compiler, if `KF_AUTOMATICVERSION` is specified. See versioning.cpp for implementation.

**Pull request merged**